mlflow/gateway/providers/openai.py

```diff
@@ -1,23 +1,111 @@
 from ..config import OpenAIConfig, RouteConfig
 
+_API_TYPE_OPENAI = "openai"
+_API_TYPE_AZURE = "azure"
+_API_TYPE_AZUREAD = "azuread"
+
 
 class OpenAIProvider(BaseProvider):
     def __init__(self, config: RouteConfig) -> None:
         super().__init__(config)
         if config.model.config is None or not isinstance(config.model.config, OpenAIConfig):
             # Should be unreachable
-            raise TypeError(f"Invalid config type {config.model.config}")
+            raise MlflowException.invalid_parameter_value(
+                "Invalid config type {config.model.config}"
+            )
         self.openai_config: OpenAIConfig = config.model.config
-        self.headers = {"Authorization": f"Bearer {self.openai_config.openai_api_key}"}
-        if org := self.openai_config.openai_organization:
-            self.headers["OpenAI-Organization"] = org
-        self.base_url = self.openai_config.openai_api_base
+        self.openai_api_type = (self.openai_config.openai_api_type or _API_TYPE_OPENAI).lower()
+        self._validate_openai_config()
+
+    def _validate_openai_config(self):
+        if not self.openai_config.openai_api_key:
+            raise MlflowException.invalid_parameter_value(
+                "OpenAI route configuration must specify 'openai_api_key'"
+            )
+
+        if self.openai_api_type == _API_TYPE_OPENAI:
+            if self.openai_config.openai_deployment_name is not None:
+                raise MlflowException.invalid_parameter_value(
+                    f"OpenAI route configuration can only specify a value for "
+                    f"'openai_deployment_name' if 'openai_api_type' is '{_API_TYPE_AZURE}' or "
+                    f"'{_API_TYPE_AZUREAD}'. Found type: '{self.openai_api_type}'"
+                )
+        elif self.openai_api_type in (_API_TYPE_AZURE, _API_TYPE_AZUREAD):
+            if self.openai_config.openai_organization is not None:
+                raise MlflowException.invalid_parameter_value(
+                    f"OpenAI route configuration can only specify a value for "
+                    f"'openai_organization' if 'openai_api_type' is '{_API_TYPE_OPENAI}'"
+                )
+
+            base_url = self.openai_config.openai_api_base
+            deployment_name = self.openai_config.openai_deployment_name
+            api_version = self.openai_config.openai_api_version
+            if (base_url, deployment_name, api_version).count(None) > 0:
+                raise MlflowException.invalid_parameter_value(
+                    f"OpenAI route configuration must specify 'openai_api_base', "
+                    f"'openai_deployment_name', and 'openai_api_version' if 'openai_api_type' is "
+                    f"'{_API_TYPE_AZURE}' or '{_API_TYPE_AZUREAD}'."
+                )
+        else:
+            raise MlflowException.invalid_parameter_value(
+                f"Invalid OpenAI API type '{self.openai_api_type}'"
+            )
```

Comment:
I think we should validate this when parsing a configuration file, not here.

Reply:
Thanks for letting me know - moved validation to the pydantic config object.